EP-1: Project Architecture:

mishokU opened the pull request:
= Refactor MVP
= Add test Rx chain

mishokU committed:
EP-1: Project Architecture:
   = Refactor MVP
   = Add test Rx chain

mishokU committed:
Merge branch 'develop' into MapFragmentRefactoring

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/di/modules/AppModule.kt:
```diff
@@ -1,21 +1,29 @@
-    fun provideMapPresenter(api : CoronavirusApi) : MapPresenter = MapPresenter(api)
+    fun provideMapPresenter(
+        api : CoronavirusApi,
+        scheduler : SchedulerProviderImpl
```
Почему шедулер через конструктор прокинут, а не засетан в самом коде например ...subscribe(Scheduler.IO)... ?

mishokU replied:
Потому что у тебя в тестовом примере так было

Aliot-EI3 replied:
вот я и спрашиваю почему. я ж те тестовы кинул не для бездумного копирования

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/base/BaseFragment.kt:
```diff
@@ -0,0 +1,32 @@
+    abstract fun getLayoutId(): Int
+
+    override fun onDestroy() {
+        super.onDestroy()
```
ты уверен, что у тебя код во всех классах отформатирован? у меня форматирует по другому

mishokU replied:
Я что-то забыл, какие горячие клавиши это делают, исправлю, хорошо

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/map/MapContract.kt:
```diff
@@ -0,0 +1,17 @@
+    interface View : BaseContract.View {
+        fun displayCountries(countries : List<String>)
+        fun showCountry(country : String)
+        fun showErrorMessage(message: String?)
```
у тебя же в базовом есть showError. возможно одна из них не нужна

mishokU replied:
Ой, да, тут я сделал избыточно, исправлю

Aliot-EI3 replied:
вопросы в контрактах вью не используй. вьюха не должна разбираться нул пришел или нет

mishokU committed:
EP-1: Project Architecture:
   = Remove data bindings
   = Refactor api to service
   = Remove one showError method

(cherry picked from commit 074d851e0736326e430c977d6d24bce666fa94d6)
